packages/sbcl/termux-tweaks.patch

```diff
@@ -0,0 +1,69 @@
+diff --git a/contrib/sb-posix/posix-tests.lisp b/contrib/sb-posix/posix-tests.lisp
+index 7c94a43..c565fad 100644
+--- a/contrib/sb-posix/posix-tests.lisp
++++ b/contrib/sb-posix/posix-tests.lisp
+@@ -527,7 +527,7 @@
+   3 #\f #\o #\o)
+ 
+ (deftest opendir.1
+-  (let ((dir (sb-posix:opendir "/")))
++  (let ((dir (sb-posix:opendir "/data/data/com.termux/cache")))
+     (unwind-protect (sb-alien:null-alien dir)
+       (unless (sb-alien:null-alien dir)
+         (sb-posix:closedir dir))))
+@@ -535,7 +535,7 @@
+ 
+ #-(and darwin x86)
+ (deftest readdir.1
+-  (let ((dir (sb-posix:opendir "/")))
++  (let ((dir (sb-posix:opendir "/data/data/com.termux/cache")))
+     (unwind-protect
+        (block dir-loop
+          (loop for dirent = (sb-posix:readdir dir)
+@@ -868,7 +868,7 @@
+ 
+ #-win32
+ (deftest mkstemp.null-terminate
+-    (let* ((default (make-pathname :directory '(:absolute "tmp")))
++    (let* ((default (make-pathname :directory '(:absolute "data/data/com.termux/cache")))
+            (filename (namestring (make-pathname :name "mkstemp-1"
+                                                 :type "XXXXXX"
+                                                 :defaults default)))
+@@ -883,9 +883,9 @@
+                                            (make-string n :initial-element #\x))
+                              :defaults default))
+           (unwind-protect
+-               (values (integerp fd) (subseq temp 0 (position #\. temp)))
++               (values (integerp fd) (subseq temp 0 (position #\. temp :from-end t)))
+             (delete-file temp))))
+-  t "/tmp/mkstemp-1")
++  t "/data/data/com.termux/cache/mkstemp-1")
+ 
+ (deftest envstuff
+     (let ((name1 "ASLIFJLSDKFJKAHGSDKLJH")
+diff --git a/src/runtime/run-program.c b/src/runtime/run-program.c
+index 929ac99..f12d21b 100644
+--- a/src/runtime/run-program.c
++++ b/src/runtime/run-program.c
+@@ -100,7 +100,7 @@ void closefrom_fallback(int lowfd)
+ {
+     int fd, maxfd;
+ 
+-#ifdef SVR4
++#if defined(SVR4) || defined(__TERMUX__)
+     maxfd = sysconf(_SC_OPEN_MAX)-1;
+ #else
+     maxfd = getdtablesize()-1;
+diff --git a/tools-for-build/grovel-headers.c b/tools-for-build/grovel-headers.c
+index 1fc1fa0..abbe071 100644
+--- a/tools-for-build/grovel-headers.c
++++ b/tools-for-build/grovel-headers.c
+@@ -40,7 +40,7 @@
+   #include <sys/times.h>
+   #include <sys/wait.h>
+   #include <sys/ioctl.h>
+-#if defined __HAIKU__ || defined __DragonFly__ || defined LISP_FEATURE_ANDROID
++#if defined __HAIKU__ || defined __DragonFly__ || defined LISP_FEATURE_ANDROID || defined __TERMUX__
```

Comment:
`LISP_FEATURE_ANDROID` isn't defined?

Reply:
Works with __ANDROID__ instead of __TERMUX__ .
Apparently LISP_FEATURE_ANDROID isn't defined. At least when building with SBCL.
Testing with ECL too now.

Reply:
I'm trying to get it to build on device for now.
Building with SBCL that was built before works, but building with ECL fails now.
```
;;; Internal error:
;;;   ** Error code 1 when executing
;;; (EXT:RUN-PROGRAM "aarch64-linux-android-clang" ("-I." "-I/data/data/com.termux/files/usr/include/" "-I/data/data/com.termux/files/usr/include" "-D_GNU_SOURCE" "-DPLATFORM_ANDROID" "-DUSE_GET_STACKBASE_FOR_MAIN" "-DIGNORE_DYNAMIC_LOADING" "-fstack-protector-strong" "-Oz" "-fPIC" "-D_THREAD_SAFE" "-Dandroid" "-O2" "-c" "/data/data/com.termux/files/usr/tmp/ecl001l5oZGt.c" "-o" "/data/data/com.termux/files/usr/tmp/ecl001l5oZGt.o")):
;;; In file included from /data/data/com.termux/files/usr/tmp/ecl001l5oZGt.c:5:
;;; In file included from /data/data/com.termux/files/usr/include/ecl/ecl-cmp.h:36:
;;; In file included from /data/data/com.termux/files/usr/include/ecl/ecl.h:22:
;;; In file included from /data/data/com.termux/files/usr/include/sys/types.h:33:
;;; In file included from /data/data/com.termux/files/usr/lib/clang/13.0.0/include/stdint.h:52:
;;; In file included from /data/data/com.termux/files/usr/include/stdint.h:32:
;;; In file included from /data/data/com.termux/files/usr/include/bits/wchar_limits.h:36:
;;; In file included from /data/data/com.termux/files/usr/include/sys/cdefs.h:371:
;;; /data/data/com.termux/files/usr/include/android/api-level.h:174:50: error: expected a platform name, e.g., 'macos'
;;; int android_get_application_target_sdk_version() __INTRODUCED_IN(24);
;;;                                                  ^
;;; /data/data/com.termux/files/usr/include/android/versioning.h:62:36: note: expanded from macro '__INTRODUCED_IN'
;;; #define __INTRODUCED_IN(api_level) __BIONIC_AVAILABILITY(introduced=api_level)
;;;                                    ^
;;; /data/data/com.termux/files/usr/include/android/versioning.h:57:71: note: expanded from macro '__BIONIC_AVAILABILITY'
;;; #define __BIONIC_AVAILABILITY(__what) __attribute__((__availability__(android,strict,__what)))
;;;                                                                       ^
;;; <command line>:6:17: note: expanded from here
;;; #define android 1
;;;                 ^
;;; In file included from /data/data/com.termux/files/usr/tmp/ecl001l5oZGt.c:5:
;;; In file included from /data/data/com.termux/files/usr/include/ecl/ecl-cmp.h:36:
;;; In file included from /data/data/com.termux/files/usr/include/ecl/ecl.h:29:
;;; In file included from /data/data/com.termux/files/usr/include/stdio.h:47:
;;; In file included from /data/data/com.termux/files/usr/include/string.h:36:
;;; /data/data/com.termux/files/usr/include/bits/strcasecmp.h:56:87: error: expected a platform name, e.g., 'macos'
;;; int strcasecmp_l(const char* __s1, const char* __s2, locale_t __l) __attribute_pure__ __INTRODUCED_IN(23);
;;;                                                                                       ^
;;; /data/data/com.termux/files/usr/include/android/versioning.h:62:36: note: expanded from macro '__INTRODUCED_IN'
;;; #define __INTRODUCED_IN(api_level) __BIONIC_AVAILABILITY(introduced=api_level)
;;;                                    ^
;;; /data/data/com.termux/files/usr/include/android/versioning.h:57:71: note: expanded from macro '__BIONIC_AVAILABILITY'
;;; #define __BIONIC_AVAILABILITY(__what) __attribute__((__availability__(android,strict,__what)))
;;;                                                                       ^
;;; <command line>:6:17: note: expanded from here
;;; #define android 1
;;;                 ^
```
Will have to try more some other day.

Reply:
This seem to be a new issue with ndk-r23, unrelated to sbcl. I can reproduce it with:

```
$ cat test.c
#include <sys/types.h>
int main() {
        return 0;
};
$ clang -Dandroid test.c
```